configure.sh

```diff
@@ -0,0 +1,203 @@
+qemu_options+=(--disable-opengl)
+qemu_options+=(--disable-sdl)
+qemu_options+=(--disable-spice)
+qemu_options+=(--disable-vte)
+
+# Disable graphical network access
+qemu_options+=(--disable-vnc)
+qemu_options+=(--disable-vnc-jpeg)
+qemu_options+=(--disable-vnc-png)
+qemu_options+=(--disable-vnc-sasl)
+
+# Disable unused filesystem support
+qemu_options+=(--disable-fdt)
+qemu_options+=(--disable-glusterfs)
+qemu_options+=(--disable-libiscsi)
+qemu_options+=(--disable-libnfs)
+qemu_options+=(--disable-libssh2)
+
+# Disable unused compression support
+qemu_options+=(--disable-bzip2)
+qemu_options+=(--disable-lzo)
+qemu_options+=(--disable-snappy)
+
+# SECURITY: Disable unused security options
+qemu_options+=(--disable-seccomp)
+qemu_options+=(--disable-tpm)
+
+# Disable userspace network access ("-net user")
+qemu_options+=(--disable-slirp)
+
+# Disable USB
+qemu_options+=(--disable-libusb)
+qemu_options+=(--disable-usb-redir)
+
+# SECURITY: Don't build a static binary (lowers security)
+# needed if qemu version is less than 2.7
+if [ ${qemu_version_major} -eq 2 ] && [ ${qemu_version_minor} -lt 7 ]; then
+	qemu_options+=(--disable-static)
+fi
+
+# Disable debug
+qemu_options+=(--disable-debug-tcg)
+qemu_options+=(--disable-qom-cast-debug)
+qemu_options+=(--disable-tcg-interpreter)
+qemu_options+=(--disable-tcmalloc)
+
+# SECURITY: Disallow network downloads
+qemu_options+=(--disable-curl)
+
+# Disable Remote Direct Memory Access (Live Migration)
+# https://wiki.qemu.org/index.php/Features/RDMALiveMigration
+qemu_options+=(--disable-rdma)
+
+# Don't build the qemu-io, qemu-nbd and qemu-image tools
+qemu_options+=(--disable-tools)
+
+# Disable XEN driver
+qemu_options+=(--disable-xen)
+
+# FIXME: why is this disabled?
+# (for reference, it's explicitly enabled in Ubuntu 17.10 and
+# implicitly enabled in Fedora 27).
+qemu_options+=(--disable-linux-aio)
+
+
+#---------------------------------------------------------------------
+# Enabled options
+
+# Enable kernel Virtual Machine support.
+# This is the default, but be explicit to avoid any future surprises
+qemu_options+=(--enable-kvm)
+
+# Required for fast network access
+qemu_options+=(--enable-vhost-net)
+
+# Always strip binaries
+# needed if qemu version is less than 2.7
+if [ ${qemu_version_major} -eq 2 ] && [ ${qemu_version_minor} -lt 7 ]; then
+	qemu_options+=(--enable-strip)
+fi
+
+# Support Ceph RADOS Block Device (RBD)
+qemu_options+=(--enable-rbd)
+
+# In "passthrough" security mode
+# (-fsdev "...,security_model=passthrough,..."), qemu uses a helper
+# application called virtfs-proxy-helper(1) to make certain 9p
+# operations safer.
+qemu_options+=(--enable-virtfs)
+qemu_options+=(--enable-attr)
+qemu_options+=(--enable-cap-ng)
+
+#---------------------------------------------------------------------
+# Other options
+case $arch in
+	 x86_64)
+		 qemu_options+=(--target-list=x86_64-softmmu)
+		 ;;
+	 arm)
+		 qemu_options+=(--target-list=arm-softmmu)
+		 ;;
+	 *)
```

Comment:
I guess you can simply have qemu_options+=(--target-list=${arch}-softmmu) and thus allow all the possible architectures in qemu target list.